algorithms/C/searching/Interpolation-search.c

```diff
@@ -0,0 +1,69 @@
+/*
+Program to search data in array using interpolation search
+Interpolation search: Used to search data in a sequence stored array
+*/
+
+#include<stdio.h>
+#include<stdlib.h>
+// Function to search data
+int interpolation_search(int *arr,int low,int high,int to_search)
+{
+    if (low<=high&&to_search>=arr[low]&&to_search<= arr[high])
+    {
+    // formula to find index
+int pos=low+(to_search-arr[low])*(high-low)/(arr[high]-arr[low]);
```

Comment:
**Possible Division by zero [CWE-369] :** 

In case the array is filled with same values as that of target ( `arr[high]` == `arr[low]` ) the denominator of the probe calculation formula will become zero.

**Fix:**
either reconstruct the probe calculating formula or handle this edge case explicitly.